draft-ietf-ppm-dap.md

```diff
@@ -654,19 +654,29 @@ configuration of each aggregator. See {{compliance}} for information on HPKE
 algorithm choices.
 
 Clients retrieve the HPKE configuration from each aggregator by
-sending an HTTP GET request to `[aggregator]/hpke_config?task_id=[task-id]`, where
+sending an HTTP GET request to `[aggregator]/hpke_config`, where
 `[aggregator]` is the aggregator's endpoint URL, obtained from the task
-parameters, and `[task-id]` is the task ID obtained from the task parameters,
+parameters. A deployment MAY choose to use different HPKE configurations
+for different tasks, so clients can optionally specify a query parameter `task_id`
+when sending an HTTP GET request to `[aggregator]/hpke_config?task_id=[task-id]`,
+where `[task-id]` is the task ID obtained from the task parameters,
 encoded in Base 64 with URL and filename safe alphabet with no padding, as
 specified in sections 5 and 3.2 of {{!RFC4648}}. If the aggregator does not
 recognize the task ID, then it responds with HTTP status code 404 Not Found and
-an error of type `unrecognizedTask`. The aggregator responds to well-formed
-requests with HTTP status code 200 OK and an `HpkeConfig` value:
+an error of type `unrecognizedTask`. If a deployment enforces different HPKE
+configurations for different tasks, and a client doesn't specify `task_id` in the
+HTTP GET request, then the aggregator responds with HTTP status code 400 Bad Request
+and an error of type `unrecognizedMessage`. The aggregator responds to well-formed
+requests with HTTP status code 200 OK and a vector of `HpkeConfig` value:
 
 [TODO: Allow aggregators to return HTTP status code 403 Forbidden in deployments
 that use authentication to avoid leaking information about which tasks exist.]
 
 ~~~
+struct {
+  HpkeConfig hpke_configs<1..2^16-1>;
+} HpkeConfigResp;
```

Comment:
I don't think this adequately addresses #248:
1. You have defined a new struct for a sequence of HPKE configs, but you have not changed the return type (see line 670 above).
2. This hasn't been discussed in the issue yet, but merely allowing the server to return multiple HPKE configs doesn't totally solve the problem. What is needed, I think, is a mechanism for allowing the client to indicate which cipher suites its supports.

In any case, I think we should push this change to a future PR. Best practices for spec review (as well as code review!) is to keep changes minimal.

Reply:
Sounds good, I reverted the changes on returning a vector of `HpkeConfig`. Previously I changed line 670 to indicate the endpoint would return a vector of `HpkeConfig` values, maybe I didn't explicitly say the return type would change to `HpkeConfigResp`.

My initial thinking was the endpoint would return all `HpkeConfig` that it supports, and clients can use any of them with a cipher suite that it recognizes. But I'm fine with addressing this in a separate PR if that's clearer.

Reply:
In practice that might decrease the likelihood of failing to negotiate an HPKE suite, but I think we can do a bit better. In any case, this can/should be handled separately.